db/model/subject.js

```diff
@@ -345,8 +349,32 @@ module.exports = function subject(seq, dataTypes) {
           }
         }
 
+        const subjectDepth =
+             inst.absolutePath.split(constants.absolutePathSeparator).length;
+
+        /*
+         * If the depth of the subject does not match the hierarchyLevel
+         * throw an error
+         */
+        if (subjectDepth !== inst.hierarchyLevel) {
+          throw new InvalidHierarchyLevel('The updated depth of ' +
+            `${inst.absolutePath}: ${subjectDepth}, does not match the ` +
+             `updated hierarchyLevel: ${inst.hierarchyLevel}`);
+        }
+
```

Comment:
Ok. I have made one final change to see if the validation will run in travis. If it does not run I will take it out from the db layer.